python/pyspark/sql/datasource.py

```diff
@@ -183,11 +186,40 @@ def streamWriter(self, schema: StructType, overwrite: bool) -> "DataSourceStream
             message_parameters={"feature": "streamWriter"},
         )
 
+    def _streamReader(self, schema: StructType) -> "DataSourceStreamReader":
+        try:
+            return self.streamReader(schema=schema)
+        except PySparkNotImplementedError:
+            return _SimpleStreamReaderWrapper(self.simpleStreamReader(schema=schema))
+
+    def simpleStreamReader(self, schema: StructType) -> "SimpleDataSourceStreamReader":
+        """
+        Returns a :class:`SimpleDataSourceStreamReader` instance for reading data.
+
+        One of simpleStreamReader() and streamReader() must be implemented for readable streaming
+        data source.
```

Comment:
Can we be more explicit about when users should choose streamReader versus simpleStreamReader here? This information will be included in the API documentation for this class.

Reply:
Added doc to SimpleStreamReader that it is only intended to light weight use case when input rate and batch size is small.